core/handler/manager_server.go

```diff
@@ -205,8 +207,8 @@ func (h *handlerManager) SetDevice(ctx context.Context, in *pb.Device) (*empty.E
 }
 
 func (h *handlerManager) DeleteDevice(ctx context.Context, in *pb.DeviceIdentifier) (*empty.Empty, error) {
-	if !in.Validate() {
-		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Device Identifier", "validation failed"))
+	if err := in.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Device Identifier", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`